graphql/database/base/__tests__/Model-properties.test.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-env jest */
+
+jest.mock('../../databaseClient')
+
+describe('BaseModel required properties', () => {
+  afterEach(() => {
+    jest.resetModules()
+  })
+
+  it('fails if "name" property is not set', () => {
+    const TestModel = require('../test-utils/ExampleModelV2').default
+    delete TestModel.name
+    expect(() => {
+      TestModel.register()
+    }).toThrow('Model is missing a required field (schema, hashKey or name)')
+  })
+
+  it('fails if "hashKey" property is not set', () => {
+    const TestModel = require('../test-utils/ExampleModelV2').default
+    delete TestModel.hashKey
+    expect(() => {
+      TestModel.register()
+    }).toThrow('Model is missing a required field (schema, hashKey or name)')
+  })
+
+  it('fails if "schema" property is not set', () => {
+    const TestModel = require('../test-utils/ExampleModelV2').default
+    delete TestModel.schema
+    expect(() => {
+      TestModel.register()
+    }).toThrow('Model is missing a required field (schema, hashKey or name)')
+  })
+
+  it('fails if "schema" property is not set', () => {
+    const TestModel = require('../test-utils/ExampleModelV2').default
+    delete TestModel.schema
+    expect(() => {
+      TestModel.register()
+    }).toThrow('Model is missing a required field (schema, hashKey or name)')
+  })
```

Comment:
Duplicate test

Reply:
For #2, we have to upgrade eslint and some other stuff. I couldn't get past it super easily but I'll work on it next.

For #3, I'm confused as to what you mean. Calling super is automatic if I don't have a constructor right? 
For methods, which would I call super in? For example, calling super in `_get` would throw a `NotImplementedException`. I also don't define the public methods in the subclasses.